src/app/main/listing/listing-form/listing-form.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { Controls, takeUntilDestroyed } from 'ngx-sub-form';
+import { tap } from 'rxjs/operators';
+import { ListingType, OneListing } from 'src/app/interfaces/listing.interface';
+import { NgxRootFormComponent } from '../../../../../projects/ngx-sub-form/src/lib/ngx-root-form.component';
+import { OneDroid } from '../../../interfaces/droid.interface';
+import { OneVehicle } from '../../../interfaces/vehicle.interface';
+import { UnreachableCase } from '../../../shared/utils';
+
+interface OneListingForm {
+  vehicleProduct: OneVehicle | null;
+  droidProduct: OneDroid | null;
+  listingType: ListingType | null;
+  id: string;
+  title: string;
+  imageUrl: string;
+  price: number;
+}
+
+@Component({
+  selector: 'app-listing-form',
+  templateUrl: './listing-form.component.html',
+  styleUrls: ['./listing-form.component.scss'],
+})
+export class ListingFormComponent extends NgxRootFormComponent<OneListing, OneListingForm>
+  implements OnInit, OnDestroy {
+  // tslint:disable-next-line:no-input-rename
+  @Input('listing')
+  public dataInput: Required<OneListing> | null = null;
+
+  // tslint:disable-next-line:no-output-rename
+  @Output('listingUpdated')
+  public dataOutput: EventEmitter<OneListing | null> = new EventEmitter();
+
+  public ListingType = ListingType;
+
+  public readonlyFormControl: FormControl = new FormControl(false);
+
+  protected getFormControls(): Controls<OneListingForm> {
+    return {
+      vehicleProduct: new FormControl(null),
+      droidProduct: new FormControl(null),
+      listingType: new FormControl(null, Validators.required),
+      id: new FormControl(null, Validators.required),
+      title: new FormControl(null, Validators.required),
+      imageUrl: new FormControl(null, Validators.required),
+      price: new FormControl(null, Validators.required),
+    };
+  }
+
+  public ngOnInit(): void {
+    super.ngOnInit();
```

Comment:
I think we should have some sort of check within the lib that `super.ngOnInit();` has been called otherwise it's easy to spend a lot of time trying to debug the form... (speaking from experience :sweat_smile:).

Reply:
See [comments here,](https://github.com/Microsoft/TypeScript/issues/21388#issuecomment-495154839) it might be a way off in TS to enforce this at compile time.

Not sure if it is a good idea or not to do something dirty with setTimeout or another hook

Reply:
For now we'll just let it like that.
I'll add it to the doc.